--- sdk.orig/C/Threads.c	2010-04-24 03:08:36.953277983 +0200
+++ sdk/Threads.c	2010-04-29 21:47:59.232488396 +0200
@@ -97,6 +97,9 @@
 
 WRes CriticalSection_Init(CCriticalSection *p)
 {
+#ifdef __MINGW32__
+  InitializeCriticalSection(p);
+#else
   /* InitializeCriticalSection can raise only STATUS_NO_MEMORY exception */
   __try
   {
@@ -104,6 +107,7 @@
     /* InitializeCriticalSectionAndSpinCount(p, 0); */
   }
   __except (EXCEPTION_EXECUTE_HANDLER) { return 1; }
+#endif
   return 0;
 }
 
