Index: pylzma/src/sdk/C/LzmaEnc.c
===================================================================
--- pylzma.orig/src/sdk/C/LzmaEnc.c
+++ pylzma/src/sdk/C/LzmaEnc.c
@@ -2092,8 +2092,9 @@ void LzmaEnc_Destroy(CLzmaEncHandle p, I
 }
 
 
-static SRes LzmaEnc_CodeOneBlock(CLzmaEnc *p, UInt32 maxPackSize, UInt32 maxUnpackSize)
+SRes LzmaEnc_CodeOneBlock(CLzmaEncHandle pp, UInt32 maxPackSize, UInt32 maxUnpackSize)
 {
+  CLzmaEnc *p = (CLzmaEnc *) pp;
   UInt32 nowPos32, startPos32;
   if (p->needInit)
   {
@@ -2526,7 +2527,7 @@ static SRes LzmaEnc_AllocAndInit(CLzmaEn
   return SZ_OK;
 }
 
-static SRes LzmaEnc_Prepare(CLzmaEncHandle pp, ISeqOutStream *outStream, ISeqInStream *inStream,
+SRes LzmaEnc_Prepare(CLzmaEncHandle pp, ISeqOutStream *outStream, ISeqInStream *inStream,
     ISzAllocPtr alloc, ISzAllocPtr allocBig)
 {
   CLzmaEnc *p = (CLzmaEnc *)pp;
@@ -2785,3 +2786,9 @@ SRes LzmaEncode(Byte *dest, SizeT *destL
   LzmaEnc_Destroy(p, alloc, allocBig);
   return res;
 }
+
+Bool LzmaEnc_IsFinished(CLzmaEncHandle pp)
+{
+  CLzmaEnc *p = (CLzmaEnc *)pp;
+  return p->finished;
+}
Index: pylzma/src/sdk/C/LzmaEnc.h
===================================================================
--- pylzma.orig/src/sdk/C/LzmaEnc.h
+++ pylzma/src/sdk/C/LzmaEnc.h
@@ -73,4 +73,11 @@ SRes LzmaEncode(Byte *dest, SizeT *destL
 
 EXTERN_C_END
 
+/* ---------- Streaming Interface ---------- */
+
+SRes LzmaEnc_Prepare(CLzmaEncHandle pp, ISeqOutStream *outStream, ISeqInStream *inStream, ISzAllocPtr alloc, ISzAllocPtr allocBig);
+SRes LzmaEnc_CodeOneBlock(CLzmaEncHandle pp, UInt32 maxPackSize, UInt32 maxUnpackSize);
+Bool LzmaEnc_IsFinished(CLzmaEncHandle pp);
+void LzmaEnc_Finish(CLzmaEncHandle pp);
+
 #endif
