diff -ur sdk.orig/C/LzmaEnc.c sdk/LzmaEnc.c
--- sdk.orig/C/LzmaEnc.c	2010-04-24 03:08:36.953277983 +0200
+++ sdk/LzmaEnc.c	2010-04-24 02:23:14.600428356 +0200
@@ -1751,8 +1751,9 @@
   alloc->Free(alloc, p);
 }
 
-static SRes LzmaEnc_CodeOneBlock(CLzmaEnc *p, Bool useLimits, UInt32 maxPackSize, UInt32 maxUnpackSize)
+SRes LzmaEnc_CodeOneBlock(CLzmaEncHandle pp, Bool useLimits, UInt32 maxPackSize, UInt32 maxUnpackSize)
 {
+  CLzmaEnc *p = (CLzmaEnc *) pp;
   UInt32 nowPos32, startPos32;
   if (p->inStream != 0)
   {
@@ -2056,7 +2057,7 @@
   return SZ_OK;
 }
 
-static SRes LzmaEnc_Prepare(CLzmaEncHandle pp, ISeqInStream *inStream, ISeqOutStream *outStream,
+SRes LzmaEnc_Prepare(CLzmaEncHandle pp, ISeqInStream *inStream, ISeqOutStream *outStream,
     ISzAlloc *alloc, ISzAlloc *allocBig)
 {
   CLzmaEnc *p = (CLzmaEnc *)pp;
@@ -2279,3 +2280,9 @@
   LzmaEnc_Destroy(p, alloc, allocBig);
   return res;
 }
+
+Bool LzmaEnc_IsFinished(CLzmaEncHandle pp)
+{
+  CLzmaEnc *p = (CLzmaEnc *)pp;
+  return p->finished;
+}
diff -ur sdk.orig/C/LzmaEnc.h sdk/LzmaEnc.h
--- sdk.orig/C/LzmaEnc.h	2010-04-24 03:08:36.953277983 +0200
+++ sdk/LzmaEnc.h	2010-04-24 02:35:27.281386039 +0200
@@ -69,4 +69,11 @@
     const CLzmaEncProps *props, Byte *propsEncoded, SizeT *propsSize, int writeEndMark,
     ICompressProgress *progress, ISzAlloc *alloc, ISzAlloc *allocBig);
 
+/* ---------- Streaming Interface ---------- */
+
+SRes LzmaEnc_Prepare(CLzmaEncHandle pp, ISeqInStream *inStream, ISeqOutStream *outStream, ISzAlloc *alloc, ISzAlloc *allocBig);
+SRes LzmaEnc_CodeOneBlock(CLzmaEncHandle pp, Bool useLimits, UInt32 maxPackSize, UInt32 maxUnpackSize);
+Bool LzmaEnc_IsFinished(CLzmaEncHandle pp);
+void LzmaEnc_Finish(CLzmaEncHandle pp);
+
 #endif
