Index: pylzma/src/sdk/C/LzmaEnc.c
===================================================================
--- pylzma.orig/src/sdk/C/LzmaEnc.c
+++ pylzma/src/sdk/C/LzmaEnc.c
@@ -2385,9 +2385,9 @@ void LzmaEnc_Destroy(CLzmaEncHandle p, I
 }
 
 
-Z7_NO_INLINE
-static SRes LzmaEnc_CodeOneBlock(CLzmaEnc *p, UInt32 maxPackSize, UInt32 maxUnpackSize)
+SRes LzmaEnc_CodeOneBlock(CLzmaEncHandle pp, UInt32 maxPackSize, UInt32 maxUnpackSize)
 {
+  CLzmaEnc *p = (CLzmaEnc *) pp;
   UInt32 nowPos32, startPos32;
   if (p->needInit)
   {
@@ -2872,7 +2872,7 @@ static SRes LzmaEnc_AllocAndInit(CLzmaEn
   return SZ_OK;
 }
 
-static SRes LzmaEnc_Prepare(CLzmaEncHandle p,
+SRes LzmaEnc_Prepare(CLzmaEncHandle p,
     ISeqOutStreamPtr outStream,
     ISeqInStreamPtr inStream,
     ISzAllocPtr alloc, ISzAllocPtr allocBig)
@@ -2883,6 +2883,12 @@ static SRes LzmaEnc_Prepare(CLzmaEncHand
   return LzmaEnc_AllocAndInit(p, 0, alloc, allocBig);
 }
 
+BoolInt LzmaEnc_IsFinished(CLzmaEncHandle pp)
+{
+  CLzmaEnc *p = (CLzmaEnc *)pp;
+  return p->finished;
+}
+
 SRes LzmaEnc_PrepareForLzma2(CLzmaEncHandle p,
     ISeqInStreamPtr inStream, UInt32 keepWindowSize,
     ISzAllocPtr alloc, ISzAllocPtr allocBig)
Index: pylzma/src/sdk/C/LzmaEnc.h
===================================================================
--- pylzma.orig/src/sdk/C/LzmaEnc.h
+++ pylzma/src/sdk/C/LzmaEnc.h
@@ -82,4 +82,11 @@ SRes LzmaEncode(Byte *dest, SizeT *destL
 
 EXTERN_C_END
 
+/* ---------- Streaming Interface ---------- */
+
+SRes LzmaEnc_Prepare(CLzmaEncHandle pp, ISeqOutStreamPtr outStream, ISeqInStreamPtr inStream, ISzAllocPtr alloc, ISzAllocPtr allocBig);
+SRes LzmaEnc_CodeOneBlock(CLzmaEncHandle pp, UInt32 maxPackSize, UInt32 maxUnpackSize);
+BoolInt LzmaEnc_IsFinished(CLzmaEncHandle pp);
+void LzmaEnc_Finish(CLzmaEncHandle pp);
+
 #endif
