Index: pylzma/src/sdk/C/CpuArch.h
===================================================================
--- pylzma.orig/src/sdk/C/CpuArch.h
+++ pylzma/src/sdk/C/CpuArch.h
@@ -325,9 +325,9 @@ int x86cpuid_GetFirm(const Cx86cpuid *p)
 #define x86cpuid_GetModel(ver)  (((ver >> 12) &  0xF0) | ((ver >> 4) & 0xF))
 #define x86cpuid_GetStepping(ver) (ver & 0xF)
 
-BoolInt CPU_Is_InOrder();
-BoolInt CPU_Is_Aes_Supported();
-BoolInt CPU_IsSupported_PageGB();
+BoolInt CPU_Is_InOrder(void);
+BoolInt CPU_Is_Aes_Supported(void);
+BoolInt CPU_IsSupported_PageGB(void);
 
 #endif
 
